### PR TITLE
[Refactor] decouple wallet rpc commands

### PR DESCRIPTION
**Pull request by jonasschnelli**
includes #5744

### CODE REVIEW

**Comment by jonasschnelli** [comment]
I think this PR's implementation is not perfect and is not very modular. But it would give the opportunity to slowly decouple the wallet from the rest of the core.
It's unclear why we have a runtime linking of rpc commands and there are probably way better solutions of a dynamic handling of the rpc commands dispatching.
But these are out of the PRs focus.

Longterm goals to achieve with this PR:
* Remove/reduce wallet_enabled #ifdefs
* Possibility to switch wallet implementation on compile and/or runtime (as startup/config arg)

File: src/rpcclient.h [hunk 1]
```diff
@@ -10,6 +10,16 @@
 #include "json/json_spirit_utils.h"
 #include "json/json_spirit_writer_template.h"
 
+/**
+ * Add new conversion to the JSON RPC conversion table
+ * @param method name of the method
+ * @param idx    0-based index of the paramter
```
Comment (fanquake) [inline src/rpcclient.h]: s/paramter/parameter

**Comment by jonasschnelli** [comment]
Rebased

**Comment by jtimon** [comment]
Needs rebase

**Comment by jonasschnelli** [comment]
@jtimon: will do soon. But maybe this PR #6006 makes more sense. What do you think?

This PR would need a better locking mechanism.

**Comment by jonasschnelli** [comment]
Rebased.
Misses a clever locking for `CRPCTable tableRPC`.
The signaling solution (https://github.com/bitcoin/bitcoin/pull/6006) which supersedes this PR is better IMO.

**Comment by jonasschnelli** [comment]
Closing in favor of #6006

**Pull request closed without merging**